app/scripts/controllers/metametrics.js

```diff
@@ -449,11 +449,16 @@ export default class MetaMetricsController {
    */
   async setParticipateInMetaMetrics(participateInMetaMetrics) {
     let { metaMetricsId } = this.state;
-    if (participateInMetaMetrics && !metaMetricsId) {
```

Comment:
`metametricsId` is no longer nullified when metametrics collection is toggled off, so we remove this condition accordingly. (see [PR](https://github.com/MetaMask/metamask-extension/pull/22162))